src/ShapeFile.php

```diff
@@ -374,7 +374,7 @@ private function loadHeaders(): bool
         $this->readSHP(4);
 
         $shapeType = Util::loadData('V', $this->readSHP(4));
-        $this->shapeType = $shapeType === false ? -1 : (int) $shapeType;
+        $this->shapeType = ShapeType::tryFrom((int) $shapeType) ?? ShapeType::Unknown;
```

Comment:
You should keep the strict check for `false`, because type casting `false` to `int` will convert it to `0`, which means `ShapeType::Null` and not `ShapeType::Unknown`.